[whilst modifying settings] Uncaught AssertionError: You must pass either an "anonymousId" or a "userId".

Reproduction Steps 
1. Go to modify sync settings
2. Blank the "analytics user id".

In this case I was using a fresh atom install, and had just unticked the "Analytics" checkbox prior to modifying the analytics user id.

My expectation would be that blanking the field (or malformed identifiers) are handled in some way.

**Atom Version**: 1.3.2
**System**: Linux Mint
**Thrown From**: [sync-settings](https://github.com/Hackafe/atom-sync-settings) package, v0.6.0
### Stack Trace

Uncaught AssertionError: You must pass either an "anonymousId" or a "userId".

```
At assert.js:89

AssertionError: You must pass either an "anonymousId" or a "userId".
    at Analytics.identify (/home/toby/.atom/packages/sync-settings/node_modules/analytics-node/lib/index.js:56:3)
    at /home/toby/.atom/packages/sync-settings/lib/tracker.coffee:42:18
    at /usr/share/atom/resources/app.asar/src/config.js:607:16
    at /usr/share/atom/resources/app.asar/src/config.js:624:20
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at Config.module.exports.Config.emitChangeEvent (/usr/share/atom/resources/app.asar/src/config.js:809:29)
    at Config.module.exports.Config.setRawValue (/usr/share/atom/resources/app.asar/src/config.js:601:19)
    at Config.module.exports.Config.set (/usr/share/atom/resources/app.asar/src/config.js:227:14)
    at SettingsPanel.module.exports.SettingsPanel.set (/usr/share/atom/resources/app.asar/node_modules/settings-view/lib/settings-panel.js:196:28)
    at /usr/share/atom/resources/app.asar/node_modules/settings-view/lib/settings-panel.js:242:26
    at Emitter.module.exports.Emitter.emit (/usr/share/atom/resources/app.asar/node_modules/event-kit/lib/emitter.js:86:11)
    at /usr/share/atom/resources/app.asar/node_modules/text-buffer/lib/text-buffer.js:1314:25
```
### Commands

```
  2x -2:06.9.0 core:paste (atom-text-editor#sync-settings.gistId.editor.mini.is-focused)
  2x -0:41.2.0 core:backspace (atom-text-editor#sync-settings.personalAccessToken.editor.mini.is-focused)
```
### Config

``` json
{
  "core": {},
  "sync-settings": {
    "gistId": "xxxxxxxxxxxxxxxxxxxx",
    "personalAccessToken": "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
    "analytics": false
  }
}
```
### Installed Packages

``` coffee
# User
sync-settings, v0.6.0

# Dev
No dev packages
```

--- a/lib/tracker.coffee
+++ b/lib/tracker.coffee
@@ -37,16 +37,32 @@ class Tracker
           version: pkg.version
         userAgent: navigator.userAgent
 
-    # identify the user
-    atom.config.observe @analyticsUserIdConfigKey, (userId) =>
-      @analytics.identify
-        userId: userId
-      @defaultEvent.userId = userId
-
     # cache enabled and watch for changes
     @enabled = atom.config.get @analyticsEnabledConfigKey
     atom.config.onDidChange @analyticsEnabledConfigKey, ({newValue}) =>
       @enabled = newValue
+      if @enabled
+        # avoid calling identify without a userId
+        if @defaultEvent.userId
+          @analytics.identify
+            userId: @defaultEvent.userId
+        else
+          uuid = require 'node-uuid'
+          # this will retrigger the observe callback and identify the userId
+          atom.config.set @analyticsUserIdConfigKey, uuid.v4()
+
+    # identify the user
+    atom.config.observe @analyticsUserIdConfigKey, (userId) =>
+      @defaultEvent.userId = userId
+      if @enabled
+        # avoid calling identify without a userId
+        if userId
+          @analytics.identify
+            userId: userId
+        else
+          uuid = require 'node-uuid'
+          # this will retrigger the observe callback and identify the userId
+          atom.config.set @analyticsUserIdConfigKey, uuid.v4()
 
   track: (message) ->
     return if not @enabled

